app/src/sharedTest/java/org/oppia/android/app/home/HomeActivityTest.kt

```diff
@@ -208,83 +208,104 @@ class HomeActivityTest {
     launch<HomeActivity>(createHomeActivityIntent(internalProfileId)).use {
       testCoroutineDispatchers.runCurrent()
-      onView(isRoot()).perform(orientationLandscape())
       scrollToPosition(position = 0)
       verifyExactTextOnHomeListItemAtPosition(
         itemPosition = 0,
-        targetViewId = R.id.profile_name_text_view,
-        stringToMatch = "Admin!"
+        targetViewId = R.id.welcome_text_view,
+        stringToMatch = "Good afternoon,Admin!"
       )
     }
   }
 
   @Test
-  fun testHomeActivity_morningTimestamp_goodMorningMessageIsDisplayed() {
+  fun testHomeActivity_Afternoon_AdminProfile_configChange_AfternoonWithProfileNameIsDisplayed() {
     fakeOppiaClock.setFakeTimeMode(FakeOppiaClock.FakeTimeMode.MODE_FIXED_FAKE_TIME)
-    fakeOppiaClock.setCurrentTimeToSameDateTime(MORNING_TIMESTAMP)
-    launch<HomeActivity>(createHomeActivityIntent(internalProfileId1)).use {
+    fakeOppiaClock.setCurrentTimeToSameDateTime(AFTERNOON_TIMESTAMP)
+    launch<HomeActivity>(createHomeActivityIntent(internalProfileId)).use {
       testCoroutineDispatchers.runCurrent()
+      onView(isRoot()).perform(orientationLandscape())
       scrollToPosition(position = 0)
       verifyExactTextOnHomeListItemAtPosition(
         itemPosition = 0,
         targetViewId = R.id.welcome_text_view,
-        stringToMatch = "Good morning,"
+        stringToMatch = "Good afternoon,Admin!"
       )
     }
   }
 
   @Test
-  fun testHomeActivity_afternoonTimestamp_goodAfternoonMessageIsDisplayed() {
+  fun testHomeActivity_withEveningTime_withAdminProfile_goodEveningWithProfileNameIsDisplayed() {
     fakeOppiaClock.setFakeTimeMode(FakeOppiaClock.FakeTimeMode.MODE_FIXED_FAKE_TIME)
-    fakeOppiaClock.setCurrentTimeToSameDateTime(AFTERNOON_TIMESTAMP)
-    launch<HomeActivity>(createHomeActivityIntent(internalProfileId1)).use {
+    fakeOppiaClock.setCurrentTimeToSameDateTime(EVENING_TIMESTAMP)
+    launch<HomeActivity>(createHomeActivityIntent(internalProfileId)).use {
       testCoroutineDispatchers.runCurrent()
       scrollToPosition(position = 0)
       verifyExactTextOnHomeListItemAtPosition(
         itemPosition = 0,
         targetViewId = R.id.welcome_text_view,
-        stringToMatch = "Good afternoon,"
+        stringToMatch = "Good evening,Admin!"
       )
     }
   }
 
   @Test
-  fun testHomeActivity_eveningTimestamp_goodEveningMessageIsDisplayed() {
+  fun testHomeActivity_withEvening_AdminProfile_configChange_EveningWithProfileNameIsDisplayed() {
     fakeOppiaClock.setFakeTimeMode(FakeOppiaClock.FakeTimeMode.MODE_FIXED_FAKE_TIME)
     fakeOppiaClock.setCurrentTimeToSameDateTime(EVENING_TIMESTAMP)
-    launch<HomeActivity>(createHomeActivityIntent(internalProfileId1)).use {
+    launch<HomeActivity>(createHomeActivityIntent(internalProfileId)).use {
       testCoroutineDispatchers.runCurrent()
+      onView(isRoot()).perform(orientationLandscape())
       scrollToPosition(position = 0)
```

Comment:
a testCouroutinesDispachers should be used before this line as we are rotating the activity which will destroy the activity and while it is being recreated it'll take some time

Reply:
does the above line (testCoroutineDispatchers.runCurrent()) does not do that, and if not can u provide me with an example of how to do this

Reply:
> a testCouroutinesDispachers should be used before this line as we are rotating the activity which will destroy the activity and while it is being recreated it'll take some time

does the above line (testCoroutineDispatchers.runCurrent()) does not do that, and if not can u provide me with an example of how to do this

Reply:
it can do that but it waits for the first time i.e. loading of the activity but we need it again as we are rotating the device

Reply:
![image](https://user-images.githubusercontent.com/76404817/155583889-4a86e9cc-0184-46bf-89d6-8b7aa24bdb55.png)
like in this test

Reply:
> ![image](https://user-images.githubusercontent.com/76404817/155583889-4a86e9cc-0184-46bf-89d6-8b7aa24bdb55.png) like in this test

I am also using a testCoroutineDispatchers.runCurrent() a line before changing orientation like in provided test example

Reply:
I haven't written this test but I also prefer to use it after a orientation change so that it can give views time to load

Reply:
> I haven't written this test but I also prefer to use it after a orientation change so that it can give views time to load

ok I will do it for all the tests which include orientation change